src/view/gui/fMainWindow.cpp

```diff
@@ -10319,3 +10319,23 @@ void fMainWindow::OnPreferencesMenuClicked()
 {
 	int result = this->preferenceDialog->exec();
 }
+
+void fMainWindow::changeEvent(QEvent *event)
+{
+	// Check window state change to ensure identical cross-platform behavior of dockwidget
+	if (event->type() == QEvent::WindowStateChange)
+	{
+		if (isMinimized()) {
+			// it was minimised
+			this->m_toolTabdock->setVisible(false);
+		}
+		else {
+			// it's normal or maximised
+			this->m_toolTabdock->setVisible(true);
```

Comment:
1) Do we need both setvisible(true) and show()?
2) This needs to happen only when dock is floating. 
3) Since this has to do with dockwidget, I would rather move as much of this as possible to captkdockwidget class.

Reply:
Let me see if the widget flag fixes this. If it does we don't have to worry about this, if not I'll fix these.

Reply:
Unfortunately, the Widget flag didn't help.

1. Not necessarily. But this issue is based upon different behavior among window managers -- i felt it would be appropriate to be as explicit as possible to help cover edge cases.
2. I can add this check but I don't think we should. If it isn't floating it's minimized alongside the window in the cases I have tried, but in Dimitri's case from the original issue he mentions just opening captk, minimizing it and seeing the dock still there. The current solution would cover this, but if we add this check it won't.

3. For odd reasons, notably that QMainWindow actually derives from _QWidget_, rather than QWindow, this unfortunately isn't as easy as it could be... nevertheless, to be more extensible, I added some small event handlers to fMainWindow that now emit signals (OnWindowMinimized, OnWindowShown). The CaPTkDockWidget class now connects to these signals in its constructor. Check out the changes when you have time.

Reply:
1. You only need one of them. 
2. isFloating check is needed. When it is not floating there is no issue at all. The issue only happens when it is floating. In case of Dimitri, it would have automatically opened the dock as floating.
3. Yeah, I agree with you. This is looking more convoluted than your earlier solution which was cleaner. I thought we could completely move it to captkdockwidget thus not having to add to the 10k lines of fmainwindow.